File: 739-daily-temperatures/daily-temperatures.py
class Solution(object):
    def dailyTemperatures(self, temperatures):
        """
        :type temperatures: List[int]
        :rtype: List[int]
        """
        stack = [] # [temp, index]
        res = [0] * len(temperatures)
        for i, temp in enumerate(temperatures):
            while stack and temp > stack[-1][0]:
                stackTemp, stackIndex = stack.pop()
                res[stackIndex] = (i - stackIndex)
            stack.append([temp, i])
        return res
        
        

File: 739-daily-temperatures/daily-temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        int n = temperatures.Length;
        int[] res = new int[n];
        Stack<(int temp, int index)> st = new Stack<(int temp, int index)>();

        for (int i = 0; i < n; i++){
            int temperature = temperatures[i];
            while (st.Count != 0 && temperature > st.Peek().temp){
                var (stTemp, stIndex) = st.Pop();
                res[stIndex] = i - stIndex;
            }
            st.Push((temperature, i));
        }
        return res;
        
    }
}